File: frontend/src/types/RoleDto.ts
import {AuthorityDto} from "./AuthorityDto.ts";

export interface RoleDto {
    id: string;

    resource: string;
    name: string;
    description: string;

    createdAt: Date;
    updatedAt: Date;
}

export interface RoleEditDto {
    resource?: string;
    name?: string;
    description?: string;
    authorityIds: string[];
}

export interface RoleWithAuthoritiesDto {
    role: RoleDto;
    authorities: AuthorityDto[];
}

export const roleToOption = (role: RoleDto) => {
    return {
        value: role.id,
        label: `${role.resource}:${role.name}`,
    };
};

File: backend/src/main/java/org/example/backend/dtos/RoleDto.java
package org.example.backend.dtos;

import lombok.Data;

import java.time.Instant;

@Data
public class RoleDto {

    private String id;

    private String resource;
    private String name;
    private String description;

    private Instant createdAt;
    private Instant updatedAt;
}
